client/cli/src/params/network_params.rs

```diff
@@ -173,6 +180,13 @@ impl NetworkParams {
 			|| is_dev
 			|| matches!(chain_type, ChainType::Local | ChainType::Development);
 
+		let allow_private_ipv4 = match (self.allow_private_ipv4, self.no_private_ipv4) {
+			(true, true) => { debug_assert!(false); true },  // The `StructOpt` library forbids that.
```

Comment:
nit: any reason not to use unreachable?
Something like that
```suggestion
			(true, true) => unreachable!("`*_private_ipv4` flags are mutually exclusive; qed"),
```